webapp/components/Button/JoinLeaveButton.vue

```diff
@@ -35,15 +36,24 @@ export default {
   },
   computed: {
     icon() {
-      if (this.isMember && this.hovered) {
-        return 'close'
+      if (this.isMember) {
+        if (this.hovered) {
+          return 'close'
+        } else if (this.isNonePendingMember) {
+          return 'check'
+        } else {
+          return 'question-circle'
+        }
```

Comment:
I don't like if their is so much depth of if else..
Maybe change the multiple return with a variable that is returned at the end.
```suggestion
      if (this.isMember && this.hovered) {
        return 'close'
      } else if (this.isMember && this.isNonePendingMember) {
        return 'check'
      } else if (this.isMember) {
        return 'question-circle'
```

Reply:
I know, on the first glimpse,it looks a bit much.
For me that clearly depicted the logic - which gets you quicker to a decision bewteen `isMember` being `true` or `false`.
The second if/else level is taking care of the three cases, if `isMember` is `true`.
Amnot sure, how much a variable would help the logic.